src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,47 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gapBetweenCell;
+  padding: 10px;
+  width: $cellSize * 7 + $gapBetweenCell * 6;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    height: $cellSize;
+    width: $cellSize;
+    box-sizing: border-box;
+    border: 1px solid #000;
+    background-color: #eee;
+
+    transition: transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+
+    @for $dayNumber from 1 through $monthLength {
+      &:nth-child(#{$dayNumber})::before {
+        content: "#{$dayNumber}";
+        font-size: 30px;
+      }
+    }
+  }
+
+  @for $hiddenDays from 28 through $monthLength {
```

Comment:
The loop should start from 1 instead of 28 to handle all possible month lengths.